src/main.rs

```diff
@@ -11,5 +14,7 @@ async fn main() -> io::Result<()> {
         server.await?;
     }
 
+    run_ws_transport().await;
```

Comment:
what do u think about sth like this?

```Rust
async fn main() -> io::Result<()> {
    let app_data = get_app_data(None).await;

    tokio::spawn(async move {
        let server = run_service(app_data);
        match server {
            Ok(server) => server.await,
            Err(e) => Err(io::Error::new(io::ErrorKind::SomeValidErrorKind, e)),
        }
    });

    tokio::spawn(async move {
        run_ws_transport().await;
    });

    Ok(())
}
```

Reply:
```Rust
async fn main() -> io::Result<()> {
    let app_data = get_app_data(None).await;

    let server_handle = tokio::spawn(async move {
        let server = run_service(app_data);
        match server {
            Ok(server) => server.await,
            Err(e) => Err(io::Error::new(io::ErrorKind::SomeValidErrorKind, e)),
        }
    });

    let ws_handle = tokio::spawn(async move {
        run_ws_transport().await;
    });

     join!(server_handle, ws_handle); // this will await on both spwaned tasks. 
//   you can also do:
//   server_handle.await;
//   ws_handle.await;

    Ok(())
}
```